File: cpp/Motherboard.cpp
#ifndef MOTHERBOARD_CPP
#define MOTHERBOARD_CPP

#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Motherboard : public Komponen {
private:
    string chipset;
    string socketType;
    int maxRAMSlots;
    string formFactor;
public:
    Motherboard() {}

    Motherboard(string chipset, string socketType, int maxRAMSlots, string formFactor, string merk, string nama) 
        : Komponen(merk, nama) {
        this->chipset = chipset;
        this->socketType = socketType;
        this->maxRAMSlots = maxRAMSlots;
        this->formFactor = formFactor;
    }

    void setChipset(string chipset) {
        this->chipset = chipset;
    }

    void setSocketType(string socketType) {
        this->socketType = socketType;
    }

    void setMaxRAMSlots(int maxRAMSlots) {
        this->maxRAMSlots = maxRAMSlots;
    }

    void setFormFactor(string formFactor) {
        this->formFactor = formFactor;
    }

    string getChipset() {
        return this->chipset;
    }

    string getSocketType() {
        return this->socketType;
    }

    int getMaxRAMSlots() {
        return this->maxRAMSlots;
    }

    string getFormFactor() {
        return this->formFactor;
    }

    void displayInfo() override {
        Komponen::displayInfo();
        cout << " (Chipset: " << chipset << ", Socket: " << socketType << ", " 
             << maxRAMSlots << " RAM Slots, " << formFactor << ")";
    }

    ~Motherboard() {}
};

#endif // MOTHERBOARD_CPP

File: python/Motherboard.py
from Komponen import Komponen

class Motherboard(Komponen):
    def __init__(self, chipset="", socket_type="", max_ram_slots=0, form_factor="", merk="", nama=""):
        super().__init__(merk, nama)
        self.chipset = chipset
        self.socket_type = socket_type
        self.max_ram_slots = max_ram_slots
        self.form_factor = form_factor
    
    def set_chipset(self, chipset):
        self.chipset = chipset
    
    def set_socket_type(self, socket_type):
        self.socket_type = socket_type
    
    def set_max_ram_slots(self, max_ram_slots):
        self.max_ram_slots = max_ram_slots
    
    def set_form_factor(self, form_factor):
        self.form_factor = form_factor
    
    def get_chipset(self):
        return self.chipset
    
    def get_socket_type(self):
        return self.socket_type
    
    def get_max_ram_slots(self):
        return self.max_ram_slots
    
    def get_form_factor(self):
        return self.form_factor
    
    def display_info(self):
        super().display_info()
        print(f" (Chipset: {self.chipset}, Socket: {self.socket_type}, {self.max_ram_slots} RAM Slots, {self.form_factor})", end="")